pytorch_lightning/utilities/data.py

```diff
@@ -406,16 +416,40 @@ def _replace_dataloader_init_method() -> Generator[None, None, None]:
         del subclass._old_init
 
 
-def _apply_fault_tolerant_automatic_capture_dataset_wrapper(dl_kwargs: Dict) -> Dict:
-    dataset = dl_kwargs["dataset"]
+def _apply_fault_tolerant_automatic_capture_dataset_wrapper(dataset: Dataset) -> Dataset:
```

Comment:
Let's rename it to something simpler: `_wrap_with_capture_dataset`

Reply:
Done